### PR TITLE
k8sgpt: init at 0.3.5

### PR DESCRIPTION
**Pull request by kranurag7**
###### Description of changes

k8sgpt is a tool for scanning your Kubernetes clusters, diagnosing, and triaging issues in simple English.
https://github.com/k8sgpt-ai/k8sgpt

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kranurag7** [comment]
Hey @IndeedNotJames while editing locally, I made sure that there is no extra lines at the end of default.nix file but in the GitHub UI, it's visible. 
I got this error message from the [workflow](https://github.com/NixOS/nixpkgs/actions/runs/4891618409/jobs/8732357252?pr=230061). 
```bash
pkgs/applications/networking/cluster/k8sgpt/default.nix:
	Wrong line endings or new final newline
```
Can you please guide me here if there's something I'm missing w.r.t tabs or spaces possibly? 
I will wait for the review before I rebase.

**Comment by IndeedNotJames** [comment]
@kranurag7 you actually need to add an empty new line at the end of the file :)

The error message might be a bit confusing.

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (IndeedNotJames) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:16]: Any particular reason why you chose to disable `CGO_ENABLED`? :)

In nixpkgs, CGO is usually kept enabled.
See https://github.com/NixOS/nixpkgs/pull/224819#issuecomment-1500890578

Comment (kranurag7) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:16]: Mostly it's about what the upstream project is following. I don't have any preference for CGO unless it's required.

**Review by wegank — APPROVED** [review]

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (IndeedNotJames) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:16]: I think I would prefer it if you could remove `CGO_ENABLED = 0;` so we can keep the nixpkgs default :)

But no worries if you don't want to.

Not blocking^^

**Comment by IndeedNotJames** [comment]
@developer-guy: @AlexsJones is still missing from [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix)^

**Comment by developer-guy** [comment]
@AlexsJones can you please provide your details in the following form?
```
developer-guy = {
  name = "Batuhan Apaydın";
  email = "developerguyn@gmail.com";
  github = "developer-guy";
  githubId = 16693043;
};
```

**Comment by kranurag7** [comment]
> can you please add us as maintainers to the package @kranurag7 along with @AlexsJones?

Sure, I will update the PR. 

> @AlexsJones can you please provide your details in the following form?

Invoking `gh api /users/AlexsJones` 
```bash
AlexsJones = {
  name = "Alex Jones";
  email = "alex@k8sgpt.ai";
  github = "AlexsJones";
  githubId = 1235925;
};
```
I will wait for Alex for the confirmation of email and other fields. 

> @developer-guy: @AlexsJones is still missing from [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix)^

I will update the commits in this [order](https://github.com/NixOS/nixpkgs/pull/222164/commits) which means I will first add Alex as the maintainer in the first commit and then `k8sgpt init` in the subsequent commit.

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:16]: I would use nixpkgs default approach, unless that breaks the program itself. Meaning not to disable CGO_ENABLED.

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:22]:
```suggestion
```

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:5]: As new version release allow me to suggest newest version changes.

```suggestion
  version = "0.3.5";
```

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:11]:
```suggestion
    sha256 = "sha256-AUUcC9S1+83ZlOH/LEG+QXstxpr8949JbqFMF9hIiVE=";
```

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:14]:
```suggestion
  vendorSha256 = "sha256-tXLt18ryxO9Q7E/MJWDlZy1gmxnMauqfcbmRdF9uUt8=";
```

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:26]:
```suggestion
    homepage = "https://k8sgpt.ai";
    changelog = "https://github.com/k8sgpt-ai/k8sgpt/releases/tag/v${version}";
```

**Review by ivankovnatsky — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:20]:
```suggestion
    "-X main.version=v${version}"
    "-X main.commit=${src.rev}"
    "-X main.date=1970-01-01-00:00:01"
```

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (kranurag7) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:14]: @ivankovnatsky Thanks for updating the hash, but I `vendorHash` is preferred. Please check this [comment](https://github.com/NixOS/nixpkgs/pull/222164#discussion_r1142076424)

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (kranurag7) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:20]: @ivankovnatsky I'm not getting the date part. Is this how we specify the date while packaging in nix? Will it consider the time at which the package is being built?

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (kranurag7) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:16]: > I would use nixpkgs default approach, unless that breaks the program itself. Meaning not to disable CGO_ENABLED.

@developer-guy, Your thoughts on this?

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (emilylange) [inline pkgs/applications/networking/cluster/k8sgpt/default.nix:11]:
```suggestion
    hash = "sha256-AUUcC9S1+83ZlOH/LEG+QXstxpr8949JbqFMF9hIiVE=";
```

Typo. Fixes eval :)

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/k8sgpt/default.nix
Comment (ivankovnatsky) [reply pkgs/applications/networking/cluster/k8sgpt/default.nix:20]: I think in some of my PRs Sandro suggest me to use this, I guess that's for compatibility purposes.

**Review by ivankovnatsky — APPROVED** [review]

**Review by emilylange — APPROVED** [review]

**Pull request merged**
